R/moduleInstall.R

```diff
@@ -1,12 +1,338 @@
+postInstallFixes <- function(folderToFix) {
+  if(length(ls(all.names=TRUE,pattern="\\..postProcessLibraryModule")) > 0 ) #We seem to be running in JASP
+  {
+    #print("we are *in* jasp, so we use .postProcessLibraryModule!")
+    .postProcessLibraryModule(folderToFix)
+  }
+  else 
+  {
+    #We do not have that function available so we will need to start JASPEngine ourselves, but where is it?
+    jaspEngineLocation <- ifElse(Sys.getenv("JASPENGINE_LOCATION") != "", Sys.getenv("JASPENGINE_LOCATION"), paste0(getwd(), "/../JASPEngine"))
+    jaspEngineCall     <- paste0(jaspEngineLocation, ' "', folderToFix ,'"')
+    #print(paste0("Calling JASPEngine as: '", jaspEngineCall ,"'"))
+    system(jaspEngineCall)
+  }
+}
+
+installJaspModule <- function(modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg, force = FALSE) {
+  assertValidJASPmodule(modulePkg)
+
+  r <- getOption("repos")
+  r["CRAN"] <- repos
+  options(repos = r)
+
+  if (!(force || md5SumsChanged(modulePkg, moduleLibrary))) {
+    moduleName <- getModuleInfo(modulePkg)[["Package"]]
+    if (dir.exists(file.path(moduleLibrary, moduleName))) {
+      print(sprintf("Nothing changed according to md5sums, not reinstalling %s.", moduleName))
+      return("succes!")
+    } else {
+      print(sprintf("Checksums exist for %s but the package is missing, installing anyway!", moduleName))
+    }
+  }
+
+  return(
+    pkgbuild::with_build_tools( 
+    {
+      if (hasRenvLockFile(modulePkg)) installJaspModuleFromRenv(       modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg)
+      else                            installJaspModuleFromDescription(modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg)
+    }, 
+    required=FALSE )
+  )
+}
+
+
+
+installJaspModuleFromRenv <- function(modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg, prompt = interactive()) {
+
+  print(sprintf("Installing module with renv. installJaspModuleFromRenv('%s', c(%s), '%s', '%s', %s)",
+                modulePkg, paste0("'", libPathsToUse, "'", collapse = ", "), moduleLibrary, repos, onlyModPkg))
+
+  renv::consent() #Consent to doing stuff in certain paths: https://rstudio.github.io/renv/reference/consent.html We've changed at least the root and the cache so hopefully that means it won't be changing stuff in the locations it mentions
+  options(renv.verbose=TRUE) # More feedback wanted although this seems to do little
+  options(renv.config.install.verbose=TRUE)
+
+  if (!dir.exists(moduleLibrary))
+    if (!dir.create(moduleLibrary))
+      stop("failed to create moduleLibrary!")
```

Comment:
Something I found out only recently is that all strings passed to `stop` and `warning` end up in weblate for translation... I assume that some of this info is only for logging purposes? In that case, it doesn't make sense to have a Chinese error message here. We can ensure that these messages won't be translated by adding `domain = NA` as an argument.

Reply:
Aah that is annoying yeah. This stuff is just for logging indeed.